app/daemon/daemon.go

```diff
@@ -112,105 +117,196 @@ func DownloadAndStart(serverConfig *conf.ServerConfig, userConfig *conf.Config)
 	}else {
 		log.Println("Located Daemon for version: " + releaseVersion)
-		return false, nil
 	}
 
+	return path, nil
+
 }
 
-func CheckForDaemon (serverConfig *conf.ServerConfig, userConfig *conf.Config) (string, error) {
 
-	log.Println("Checking daemon")
+func start(daemonPath string) (*exec.Cmd)  {
+	
+	log.Println("Booting deamon")
+	cmd := exec.Command(daemonPath)
+	cmd.Start()
 
-	// get the latest release info
-	releaseVersion := userConfig.RunningNavVersion
+	return cmd
 
-	log.Println("Looking for version: " + releaseVersion)
+}
 
-	// get the apps current path
-	path, err := getCurrentPath()
+
+
+// Get the current os info and the daemon name for that os
+func getOSInfo () OSInfo {
+
+	osInfo := OSInfo{}
+
+	//const goosList = "android darwin dragonfly freebsd linux nacl netbsd openbsd plan9 solaris windows zos "
+	//const goarchList = "386 amd64 amd64p32 arm armbe arm64 arm64be ppc64 ppc64le mips mipsle mips64 mips64le mips64p32 mips64p32le ppc s390 s390x sparc sparc64 "
+
+	switch runtime.GOARCH {
+
+	case "amd64":
+
+		switch runtime.GOOS {
+		case "windows":
+			osInfo.DaemonName = WindowsDaemonName
+			osInfo.OS = "win64"
+		}
+
+	}
+
+	return osInfo
+
+}
+
+// gets the current path of the go application
+func getCurrentPath() (string, error)  {
+
+	ex, err := os.Executable()
 	if err != nil {
 		return "", err
 	}
+	exPath := filepath.Dir(ex)
+	return exPath, nil
+}
 
-	// append the deamon name
 
-	path += "/lib/navcoin-" + releaseVersion + "/bin/"+ getOSInfo().DaemonName
-	log.Println("Looking for Navcoin Daemon at " + path )
+// Exists reports whether the named file or directory exists.
+func exists(name string) bool {
+	if _, err := os.Stat(name); err != nil {
+		if os.IsNotExist(err) {
+			return false
+		}
+	}
+	return true
+}
 
 
-	// check the daemon exists
-	if !exists(path) {
-		log.Println("No Daemon found for version: " + releaseVersion)
-		return "", errors.New("No Daemon found for version: " + releaseVersion)
-	}else {
-		log.Println("Located Daemon for version: " + releaseVersion)
+func downloadDaemon(serverConf *conf.ServerConfig, verson string) {
+
+	releaseInfo, _ := getReleaseDataForVersion(serverConf.ReleaseAPI, verson)
+
+	dlPath, dlName, _ := getDwLdInfoFromReleaseInfo(releaseInfo)
+
+	downloadUnzip(dlPath, dlName)
+
+}
+
+func getReleaseDataForVersion(releaseAPI string, version string) (GitHubReleaseData, error)  {
+
+	log.Println("Atempting to get release data")
+	releases, err := gitHubReleaseInfo(releaseAPI)
+
+	var e GitHubReleaseData = GitHubReleaseData{}
+	for _, elem := range releases {
+		if elem.TagName == version {
+			log.Println("Release data found for version: " + version)
+			e = elem.GitHubReleaseData
+		}
 	}
 
-	return path, nil
+	return e, err
 
 }
 
 
-func getDaemonDownloadPath(version string) {
+func gitHubReleaseInfo(releaseAPI string) (GitHubReleases, error) {
 
-	
+	log.Println("Retreving NAVCoin Github releases info from: " + releaseAPI)
```

Comment:
I think it might be better as `Retrieving NAVCoin Github releases' info from:`

Reply:
done :)